src/NSubstitute/Extensions/ReceivedExtensions.cs

```diff
@@ -155,5 +157,32 @@ public override bool Equals(object obj)
 
             public override int GetHashCode() { return 0; }
         }
+
+        private class RangeQuantity : Quantity
+        {
+            private readonly int minInclusive;
+            private readonly int maxInclusive;
+            public RangeQuantity(int minInclusive, int maxInclusive) {
+                if (minInclusive < 0) {
```

Comment:
I suggest to move this check to the `Within()` static method above. There we deal with user input, so it would be more natural to have checks there. Given that `RangeQuantity` is private, I think it's OK to skip checks here, so we can enforce consistency withing single file.

Reply:
I understand the reasoning but I think I prefer it here. Then we get the validation if we new this up elsewhere later on.

Reply:
👍